fabric-biome-api-v1/src/main/java/net/fabricmc/fabric/impl/biome/WeightedPicker.java

```diff
@@ -100,6 +101,16 @@ WeightedEntry<T> search(final double target) {
 		return entries.get(low);
 	}
 
+	/**
+	 * Returns any {@link WeightedEntry} that contains the passed element.
+	 *
+	 * @param element The Element you are looking for
+	 * @return The result of the search
+	 */
+	Optional<WeightedEntry<T>> getAny(T element){
+		return entries.stream().filter(w -> w.entry==element).findAny();
+	}
```

Comment:
Instead of this, you might want to look into changing `TheEndBiomeData#ADDED_BIOMES` into a `Map<RegistryKey<Biome>, RegistryKey<Biome>>` instead, with vanilla biome as its value.  Then you can check if the value is equal to the vanilla biome.

[Patch file](https://gist.github.com/deirn/d8eb50934e666616b2982d10817c659e)

Reply:
Yeah, that is much cleaner. I'll include the suggested patch...